components/server/src/github/github-auth-provider.ts

```diff
@@ -104,7 +104,8 @@ export class GitHubAuthProvider extends GenericAuthProvider {
                     authName: login,
                     avatarUrl: avatar_url,
                     name,
-                    primaryEmail: userEmails.filter(e => e.primary)[0].email
```

Comment:
@philschatz, I think we need to explicitly check for the primary email address to be private.

I just played with the `Keep my email addresses private` setting and the `https://api.github.com/user/emails` API, and found that if active the proxy address is included in the list. This dynamic behavior should allow to reflect the setting on each login into Gitpod. 

Could you please adopt the strategy to do this?

```
const primary = userEmails.filter(e => e.primary)[0]!;
const proxy = userEmails.find(e => e.email.endsWith(`@users.noreply.${this.config.host}`));
...
  primaryEmail: primary.email,
  commitEmail: primary.visibility === "private" ? proxy?.email : primary.email,
...
```